### PR TITLE
add validates for all model

### PR DESCRIPTION
**Pull request by linhmillion**

### CODE REVIEW

File: app/models/challenge.rb [hunk 1]
```diff
@@ -1,2 +1,18 @@
 class Challenge < ApplicationRecord
+  CATEGORY = %w[
```
Comment (VeganBt) [inline app/models/challenge.rb:2]: should change to CATEGORIES

File: app/models/group.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Group < ApplicationRecord
+  has_many :memberships
```
Comment (VeganBt) [inline app/models/group.rb:2]: should also have 
`has_many :users`

File: app/models/group_challenge.rb [hunk 3]
```diff
@@ -1,4 +1,17 @@
 class GroupChallenge < ApplicationRecord
+  STATUS = %w[waiting challenging finished]
```
Comment (VeganBt) [inline app/models/group_challenge.rb:2]: should be plural STATUSES

File: app/models/membership.rb [hunk 4]
```diff
@@ -1,4 +1,9 @@
 class Membership < ApplicationRecord
+  MEMBER_STATUS = %w[Pending Accept Declined]
```
Comment (VeganBt) [inline app/models/membership.rb:2]: also should be plural MEMBER_STATUSES

Comment (daanbm) [inline app/models/membership.rb:2]: Accepted*

File: app/models/user.rb [hunk 5]
```diff
@@ -3,4 +3,10 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email_address, presence: true, format: /\w+@\w+\.{1}[a-zA-Z]{2,}/
+  has_many :memberships
+  # has_one_attached :photo
```
Comment (VeganBt) [inline app/models/user.rb:11]: also 
`has_many :groups, through: :memberships`

**Review by daanbm — COMMENTED** [review]
Please see comments from Ofer and myself

File: app/models/group_challenge.rb [hunk 6]
```diff
@@ -1,4 +1,17 @@
...
   belongs_to :challenge
   belongs_to :membership
+  validates :status, presence: true, inclusion: { in: STATUS }
+  validates :start_date, presence: true
+
```
Comment (daanbm) [inline app/models/group_challenge.rb:7]: Can we have a default value for votes set at 0?

**Pull request merged**